blocks/block-controls/index.js

```diff
@@ -3,11 +3,18 @@
  */
 import { Toolbar, Fill } from '@wordpress/components';
 
-export default function BlockControls( { controls, children } ) {
+/**
+ * Internal dependencies
+ */
+import { ifEditBlockSelected } from '../block-edit/context';
+
+export function BlockControls( { controls, children } ) {
```

Comment:
How can we inform the user about this backward compatibity change. I think it's fine to ship it because users are probably already testing `isSelected` on their side but how can we inform them that it's not useful anymore?

Reply:
It isn't a breaking change, so it's more about the convenience of developers and assuring that they don't introduce subtle bug as noted in [here](https://javascriptforwp.com/forums/topic/duplicate-inspector-controls/). I don't think we can add a deprecation message in this case as it's passed down in case of `BlockControls` and `InspectorControls`.

Reply:
I was thinking maybe not a warning but we add a log message like this when registering custom blocks:

```
In this new Gutenberg Release, The `BlockControls`,  `InspectorControls` and `RichText` automatically hides when the block is not selected, you don't need to check or pass the `isSelected` prop anymore.
```

And just leave this message for one release at least? What do you think?

Reply:
Yes, I like this idea. I didn't think of it this way :)